docs/architecture/driver.rst

```diff
@@ -113,7 +113,14 @@ Example code to implement a writer driver
         def format(self):
             return ''  # Format that this writer supports
 
-        def write_dataset_to_storage(self, dataset, filename,
+        def mk_uri(self, file_path, storage_config):
+            """
+            Constructs a uri from the file_path and storage config.
```

Comment:
Can we expand a bit on what `file_path` and `storage_config` parameters are?
Where they are coming from, what type they are.
What should a "typical" implementation that writes to local filesystem should return from this function.